controller/volume_controller.go

```diff
@@ -1402,6 +1407,19 @@ func (c *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[strin
 	return c.checkAndFinishVolumeRestore(v, e, rs)
 }
 
+func (c *VolumeController) checkVolumeMountStatus(v *longhorn.Volume, e *longhorn.Engine) (err error) {
+	log := getLoggerForVolume(c.logger, v)
+	if v.Status.State == longhorn.VolumeStateAttached && e.Status.CurrentState == longhorn.InstanceStateRunning {
+		if strings.Contains(e.Status.ErrorMsg, "READONLY") {
```

Comment:
Can we introduce `e.Status.FilesystemReadOnly` for preventing string comparison?

Reply:
Fixed, but I think it has room to discuss if we should put the value to engine status
or keep it in engine.Status.InstanceStatus.XXX, so we don't have to copy/pass the value